src/cmap/commands.ts

```diff
@@ -837,6 +838,12 @@ export class BinMsg {
     const promoteValues = options.promoteValues ?? this.opts.promoteValues;
     const promoteBuffers = options.promoteBuffers ?? this.opts.promoteBuffers;
     const bsonRegExp = options.bsonRegExp ?? this.opts.bsonRegExp;
+    if (options.validation && Object.keys(options.validation.utf8)[0] !== 'writeErrors') {
```

Comment:
Are we not allowing general validation options at this time?  The type on line 472 implies that ` { validation: true / false } ` would be allowed, but this logic would throw an error if that was passed in.

Reply:
This is an internal workaround to deal with server errors containing invalid utf8, so we're not really allowing keys besides writeErrors to have validation enabled/disabled. But you're right, I should narrow this down more (but I'm getting type errors if I try to change the type from `Record<string, true> | Record<string, false>` to `Record<string, boolean>`, since its saying the type isn't assignable to type 'DeserializeOptions' even though I omitted validation from the BSONSerializeOptions type?).

Reply:
Oh ok - so this is an internal change and not a user facing interface?  And it's building off of your work in the BSON library?  

As long as it's not user-facing, I would probably type it the way you had it originally so that it matches the types in the bson repo [here](https://github.com/mongodb/js-bson/blob/522182577733a3b7a4d70d3c509cc7e6621d0470/src/parser/deserializer.ts#L63).  But I'm also still new so I'm interested to hear what the team thinks

Reply:
@baileympearson good points here, let me try to address some of them: this validation and the things inside the class are 100% internal, we could probably make do without that validation at all, given that it's internally driven at the moment; however, we have a follow-up ticket to allow users to disable validation entirely, so I think this was Grace trying to set up the stage for the follow up work. That said, I don't think we'd want to try to validate this here, instead, we'd probably validate the user options in `resolveBSONOptions`. @gjchong25 I think this is safe to leave for the follow up work. Can we remove the validation check and the passing in of the options? Just hardcode the `const validation = { utf8: { writeErrors: false } }` for now?

Reply:
I dropped the throw / and checks